shell/platform/darwin/ios/framework/Source/spring_curve_ios.mm

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/spring_curve_ios.h"
+
+#include <Foundation/Foundation.h>
+
+// This simplified spring model is based off of a damped harmonic oscillator.
+// See: https://en.wikipedia.org/wiki/Harmonic_oscillator#Damped_harmonic_oscillator
+//
+// This models the closed form of the second order differential equation which happens to match the
+// algorithm used by CASpringAnimation, a QuartzCore (iOS) API that creates spring animations.
+//
+// Spring code adapted from React Native's Animation Library, see:
+// https://github.com/facebook/react-native/blob/main/Libraries/Animated/animations/SpringAnimation.js
```

Comment:
Nits: Maybe better to write in `.h` file instead of `.mm` file? ^_^

Reply:
I figured better to keep it with the implementation in the `.mm` file  rather than the `.h` file, but will move if you think more appropriate.

Reply:
I think the `.h` file will be a place for user for this class, maybe easier for them to see haha

Reply:
I split the comment between the two, keeping the react reference in the implementation, think that makes the most sense to me